File: Baekjoon/Stack & Queue/스택(10828).py
import sys
input = sys.stdin.readline

class Stack:
    def __init__(self):
        self.stack = []

    def push(self, X):
        self.stack.append(X)

    def pop(self):
        stack_pop = self.stack.pop() if len(self.stack)>0 else -1
        print(stack_pop)
        return stack_pop
    
    def size(self):
        stack_size = len(self.stack)
        print(stack_size)
    
    def empty(self):
        stack_empty = 0 if len(self.stack)>0 else 1
        print(stack_empty)
    
    def top(self):
        stack_top = self.stack[len(self.stack) - 1] if len(self.stack)>0 else -1
        print(stack_top)   
        
N = int(input())
stack = Stack()

S = {
    'push': stack.push,
    'pop': stack.pop,
    'size': stack.size,
    'empty': stack.empty,
    'top': stack.top
}

for n in range(N):
    cmd = input().split()
    key = cmd[0]
    value = cmd[1] if len(cmd)>1 else None
    
    if key == 'push':
        S[key](value)
        continue;
    S[key]()

File: Baekjoon/Stack & Queue/스택(10828).js
/**
 * 
 */

// class Stack {
//     #stack = [];

//     push(X) {
//         this.#stack.push(+X);
//     }

//     pop() {
//         const stackPop = this.#stack.pop() || -1;
//         console.log(stackPop);
//         return stackPop;
//     }

//     size() {
//         const stackSize = this.#stack.length;
//         console.log(stackSize);
//     }

//     empty() {
//         const isEmpty = this.#stack.length ? 0 : 1;
//         console.log(isEmpty);
//     }

//     top() {
//         const stackTop = this.#stack[this.#stack.length-1] || -1;
//         console.log(stackTop);
//     }
// }

// const stack = new Stack();

// const N = +input.shift();
// for (const cmd of input) {
//     const [key, value] = cmd.trim().split(' ');
    
//     if (key === 'push') {
//         stack[key](value);
//         continue;
//     }
//     stack[key]();
// }


const input = require('fs').readFileSync('./input').toString().split('\n');

const stack = [];
const N = input.shift();

let result = '';
for (const cmd of input) {
    const [key, value] = cmd.trim().split(' ');
    switch (key) {
        case 'pop':
            const stackPop = stack.pop() || -1;
            result += stackPop + '\n';
            break;
        case 'size':
            const stackSize = stack.length;
            result += stackSize + '\n';
            break;
        case 'empty':
            const isEmpty = stack.length ? 0 : 1;
            result += isEmpty + '\n';
            break;
        case 'top':
            const stackTop = stack[stack.length-1] || -1;
            result += stackTop + '\n';
            break;
        default:
            stack.push(value);
    }
}

console.log(result);